pyproject.toml

```diff
@@ -10,9 +10,9 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.9,<3.13"
```

Comment:
This doesn't need to change just because we are allowing cryptography 44.0.0

Reply:
The issue I'm running into is that, if I only change the upper bound of the cryptography dependency to <45.0 while keeping the lower bound at >=41.0.1 and regenerate the lock file (`poetry update cryptography --lock`), the lockfile calls for cryptography 43.0.3. 

However, if I also update the lower bound to >=44.0 to force cryptography 44.0.0, Poetry gives me the following error:

```
Updating dependencies
Resolving dependencies... (0.1s)

The current project's supported Python range (>=3.9,<3.13) is not compatible with some of the required packages Python requirement:
  - cryptography requires Python !=3.9.0,!=3.9.1,>=3.7, so it will not be satisfied for Python 3.9 || 3.9.1

Because no versions of cryptography match >44.0,<45.0
 and cryptography (44.0.0) requires Python !=3.9.0,!=3.9.1,>=3.7, cryptography is forbidden.
So, because rivian-python-client depends on cryptography (>=44.0,<45.0), version solving failed.

  • Check your dependencies Python requirement: The Python requirement can be specified via the `python` or `markers` properties
    
    For cryptography, a possible solution would be to set the `python` property to ">3.9.0,<3.9.1 || >3.9.1,<3.13"

    https://python-poetry.org/docs/dependency-specification/#python-restricted-dependencies,
    https://python-poetry.org/docs/dependency-specification/#using-environment-markers
```

I defer to you all -- I am not a developer by trade and may well be missing something -- but it struck me that the simplest solution was to update the upper bound for cryptography to <45.0 and the lower bound for Python to >3.9.1:

```
[tool.poetry.dependencies]
python = ">3.9.1,<3.13"
aiohttp = ">=3.0.0"
cryptography = ">=41.0.1,<45.0"
backports-strenum = { version = "^1.2.4", python = "<3.11" }
bleak = { version = ">=0.21,<1.0.0", optional = true }
dbus-fast = {version = "^2.11.0", optional = true, platform = "linux"}
```

Reply:
Try reverting the python version constraint and then just running `poetry lock --no-update`

Reply:
I appreciate the suggestions, but when I try that and then build/install the updated package locally to test, it's still looking for cryptography 43.0.3, which leads to the same dependency conflict that prevents the Home Assistant integration from loading, whereas if I update the Python version constraint and then build/install, the integration fires right up. 

I defer fully to you all. I'm happy to close my PR and let someone else fix the issue in whatever way you and others think makes the most sense.

Reply:
@krazos Try updating the manifest.json file in your rivian custom component to require rivian-python-client[ble]==1.3.1 and see if that works?